### PR TITLE
add Web-API documentation

### PR DESCRIPTION
**Pull request by madmartin**
please take a look, this is WIP - POST examples will follow

Signed-off-by: Martin Dummer <martin.dummer@gmx.net>

### CODE REVIEW

File: docs/Web-API.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+
+| GET/POST | URL |
+| -- | -- |
+| Get | /api/config/get |
+| Post | /api/config/delete |
+| Post | /api/config/upload |
+| Get | /api/devinfo/status |
+| Get+Post | /api/dtu/config |
+| Get | /api/eventlog/status?inv=interter-serialnumber |
+| Get | /api/firmware/update |
```
Comment (tbnobody) [inline docs/Web-API.md:18]: /api/firmware/api is a Post command

File: docs/Web-API.md [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+| Get | /api/inverter/list |
+| Post | /api/inverter/add | 
+| Post | /api/inverter/del |
+| Post | /api/inverter/edit |
+| Post | /api/limit/config |
+| Get | /api/limit/status |
+| Get | /api/livedata/status |
+| Get+Post | /api/mqtt/config |
+| Get | /api/mqtt/status |
+| Get | /api/network/status |
```
Comment (tbnobody) [inline docs/Web-API.md:28]: /api/network/config is missing and Get+Post

File: docs/Web-API.md [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+| Get+Post | /api/ntp/config |
+| Get | /api/ntp/status |
+| Get | /api/ntp/time |
```
Comment (tbnobody) [inline docs/Web-API.md:31]: /api/ntp/time is Get+Post

**Review by tbnobody — COMMENTED** [review]
The different json objects are also described in the webapp as TypeScript interfaces. `/webapp/src/types/`

**Comment by madmartin** [comment]
Hello @tbnobody ,
I tried a lot to get  some working curl post examples, but all failed. 
You may merge if you want.

**Comment by tbnobody** [comment]
@madmartin I cannot test it at the moment but you could try something like this:
`curl -X POST http://192.168.20.110/api/ntp/config -d "data={}"`
The JSON is expected to be in a FormData Format with `data` as Form input field name. It's a workaround to make handling on the C/C++ side easier.

**Comment by madmartin** [comment]
Unfortuanely not....

```
~ $ curl -X POST http://192.168.166.189/api/ntp/config -d 'data={"ntp_server":"de.pool.ntp.org"}'
{"type":"warning","message":"Values are missing!"}
```

Also the code delivered from [Postman](https://www.postman.com/) after an inspection with postman as proxy does not work.
```
~ $ curl --location -g --request POST 'http://192.168.166.189/api/ntp/config' \
--header 'host: 192.168.166.189' \
--header 'user-agent: Mozilla/5.0 (X11; Linux x86_64; rv:106.0) Gecko/20100101 Firefox/106.0' \
--header 'accept: */*' \
--header 'accept-language: de,en-US;q=0.7,en;q=0.3' \
--header 'accept-encoding: gzip, deflate' \
--header 'referer: http://192.168.166.189/settings/ntp' \
--header 'content-type: multipart/form-data; boundary=---------------------------293084418715961144331856902114' \
--header 'content-length: 279' \
--header 'origin: http://192.168.166.189' \
--header 'connection: keep-alive' \
--data-raw '-----------------------------293084418715961144331856902114
Content-Disposition: form-data; name="data"

{"ntp_server":"de.pool.ntp.org","ntp_timezone":"CET-1CEST,M3.5.0,M10.5.0/3","ntp_timezone_descr":"Europe/Berlin"}
-----------------------------293084418715961144331856902114--
'
curl: (52) Empty reply from server
```

**Comment by tbnobody** [comment]
But you are getting the message` Values are missing!` because you only supplied the `ntp_server`. The API is implemented in a way that always the whole json object is expected. 
`curl -X POST http://192.168.166.189/api/ntp/config -d 'data={"ntp_server":"de.pool.ntp.org","ntp_timezone":"CET-1CEST,M3.5.0,M10.5.0/3","ntp_timezone_descr":"Europe/Berlin"}'`

Will double check this later too.

**Comment by tbnobody** [comment]
```
$ curl -X POST http://192.168.20.110/api/ntp/config -d 'data={"ntp_server":"192.168.1.1","ntp_timezone":"CET-1CEST,M3.5.0,M10.5.0/3","ntp_timezone_descr":"Europe/Berlin"}'
{"type":"success","message":"Settings saved!"}
```

**Comment by madmartin** [comment]
@tbnobody Okay, got it now, some usage notes and post examples are added. Ready to merge from my side, or do you want more explanations on settings? (power limits, add/delete inverters)?

**Comment by madmartin** [comment]
Found a breaking change in commit 69b675bc for /api/livedata/status  - I will update the example tomorrow when I have new live data.

**Comment by madmartin** [comment]
@tbnobody examples are updated now with new live data.

**Comment by tbnobody** [comment]
Yesterday I already prepared 244c94e for merging and added some remarks regarding authentication. 
I will try to merge the diff between https://github.com/tbnobody/OpenDTU/compare/244c94e4561b0116049c855ec075d023812279b1..d30320bcfe08c6d7baa43e93f75482e013074922 on top of my current tree

**Comment by tbnobody** [comment]
I added your latest changes by hand and going to close this PR as it gets untransparent (with maybe a lot of force pushes)

**Pull request closed without merging**